cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -1,39 +1,44 @@
 package topgun.cmdline
 
 import java.io.File
-import java.nio.file.Files
-
-import com.jrockit.mc.common.IMCFrame
-import com.jrockit.mc.flightrecorder.FlightRecordingLoader
-import com.jrockit.mc.flightrecorder.internal.model.{FLRStackTrace, FLRThread, FLRType}
-import com.jrockit.mc.flightrecorder.spi.IEvent
+import java.util.stream.Collectors
+import java.util
+import jdk.jfr.consumer.{RecordedClass, RecordedEvent, RecordedFrame, RecordingFile}
 import topgun.core.CallSite
 
 import scala.jdk.CollectionConverters._
 
 class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, configuration: Configuration) {
 
+  var allocationInNewTlab: Boolean = false
+  var allocationOutsideTlab: Boolean = false
+  var methodProfilingSample: Boolean = false
+
   import configuration.{includeStack, includeThread}
 
+  def foundRequiredEvents(): Boolean = {
+    if(!allocationInNewTlab){
+      throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Allocation in new TLAB' event not found")
+    }
+    if(!allocationOutsideTlab){
+      throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Allocation outside TLAB' event not found")
+    }
+    if(!methodProfilingSample){
+      throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Method Profiling Sample' event not found")
+    }
+    return true;
+  }
+
   def parse(): Unit = {
 
     println(s"*** File $file")
     try {
-      val recording = FlightRecordingLoader.loadFile(file)
-      for (t <- recording.getEventTypes.asScala) {
-        println(t.getName)
-      }
-      val view = recording.createView
-      view.asScala.find(_.getEventType.getName == "GC TLAB Configuration") match {
-        case Some(e) => handleTlabConfiguration(e)
-        case _ => ???
-      }
-
+      val view = RecordingFile.readAllEvents(file.toPath);
```

Comment:
this reads all the events, into a `List`

can weuse an Iterator - like https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jfr/jdk/jfr/consumer/RecordingFile.html
```
while (recordingFile.hasMoreEvents()) {
     RecordedEvent event = recordingFile.readEvent();
     event match {....}
   }

```

Reply:
oh okay.

Reply:
done